konduit-serving-clients/src/main/java/ai/konduit/serving/clients/generators/GenerateRestClients.java

```diff
@@ -0,0 +1,328 @@
+import io.swagger.codegen.v3.ClientOpts;
+import io.swagger.codegen.v3.CodegenConstants;
+import io.swagger.codegen.v3.DefaultGenerator;
+import io.swagger.codegen.v3.generators.java.JavaClientCodegen;
+import io.swagger.codegen.v3.generators.python.PythonClientCodegen;
+import io.swagger.v3.core.converter.ModelConverters;
+import io.swagger.v3.core.util.Json;
+import io.swagger.v3.core.util.Yaml;
+import io.swagger.v3.oas.models.ExternalDocumentation;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.Operation;
+import io.swagger.v3.oas.models.PathItem;
+import io.swagger.v3.oas.models.info.Contact;
+import io.swagger.v3.oas.models.info.Info;
+import io.swagger.v3.oas.models.info.License;
+import io.swagger.v3.oas.models.media.*;
+import io.swagger.v3.oas.models.parameters.RequestBody;
+import io.swagger.v3.oas.models.responses.ApiResponse;
+import io.swagger.v3.oas.models.responses.ApiResponses;
+import io.swagger.v3.oas.models.tags.Tag;
+import javassist.CannotCompileException;
+import javassist.ClassPool;
+import javassist.CtClass;
+import javassist.NotFoundException;
+import javassist.bytecode.AnnotationsAttribute;
+import javassist.bytecode.ClassFile;
+import javassist.bytecode.ConstPool;
+import javassist.bytecode.annotation.*;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FileUtils;
+import org.nd4j.common.io.ClassPathResource;
+import org.nd4j.common.primitives.Pair;
+
+import java.io.*;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static io.netty.handler.codec.http.HttpHeaderValues.APPLICATION_JSON;
+import static io.netty.handler.codec.http.HttpHeaderValues.APPLICATION_OCTET_STREAM;
+
+@Slf4j
+public class GenerateRestClients {
+
+    public static void main(String[] args) throws NotFoundException, IOException {
+        // Setting this so that the Json Serializer is able see private fields without standard getter methods.
+        Json.mapper()
+                .setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY)
+                .setVisibility(PropertyAccessor.CREATOR, JsonAutoDetect.Visibility.ANY);
+
+        Map<String, List<Pair<String, String>>> mappings = getJsonNameMappings();
+        mappings.put("ai.konduit.serving.endpoint.Endpoint",
+                Collections.singletonList(new Pair<>(null, "ai.konduit.serving.endpoint.AssetServingEndpoint"))
+        );
+        mappings.put("ai.konduit.serving.pipeline.api.data.BoundingBox",
+                Arrays.asList(new Pair<>(null, "ai.konduit.serving.pipeline.impl.data.box.BBoxCHW"),
+                        new Pair<>(null, "ai.konduit.serving.pipeline.impl.data.box.BBoxXY"))
+        );
+        mappings.put("ai.konduit.serving.pipeline.api.pipeline.Pipeline",
+                Arrays.asList(new Pair<>(null, "ai.konduit.serving.pipeline.impl.pipeline.SequencePipeline"),
+                        new Pair<>(null, "ai.konduit.serving.pipeline.impl.pipeline.GraphPipeline"))
+        );
```

Comment:
Tried to find class names through reflection but it limits class modification after `Class#forName` is already called before and then it will give you `java.lang.LinkageError: attempted duplicate class definition` when you try to recreate it with the modified class (using CtClass#toClass)`. There's an easy fix though that's similar to how I'm looking for class names from a pre-generated metadata file (JsonNameMapping). We can have a separate file that contains information about some base classes and their extensions. Over here we don't really need the `@JsonName` metadata since this is just to let the swagger generator create inherited classes and allow polymorphism while creating `InferenceConfiguration` objects. I've created an issue for this over here like you suggested:   https://github.com/KonduitAI/konduit-serving/issues/369